idb/postgres/postgres_integration_test.go

```diff
@@ -31,6 +31,79 @@ func getAccounting(round uint64, cache map[uint64]bool) *accounting.State {
 	return accountingState
 }
 
+// TestEmbeddedNullString make sure we're able to import cheeky assets.
+func TestEmbeddedNullString(t *testing.T) {
+	db, shutdownFunc := setupIdb(t, test.MakeGenesis())
+	defer shutdownFunc()
+
+	assetID := uint64(1)
+	nameWithNull := "my\000coin"
+	unitWithNull := "m\000c"
+	urlWithNull := "http://its\000happening.com"
+	txn, txnRow := test.MakeAssetConfigOrPanic(
+		test.Round, 0, assetID, math.MaxUint64, 0, false, unitWithNull, nameWithNull, urlWithNull, test.AccountA)
+
+	// Test 1: import/accounting should work.
+	importTxns(t, db, test.Round, txn)
+	accountTxns(t, db, test.Round, txnRow)
+
+	// Test 2: search for weird asset with no results
+	assets, _ := db.Assets(context.Background(), idb.AssetsQuery{Name: "No\001Issu\\es"})
+	num := 0
+	for asset := range assets {
+		require.NoError(t, asset.Error)
+		num++
+	}
+	assets, _ = db.Assets(context.Background(), idb.AssetsQuery{Name: "No\000Issu\\es"})
+	for asset := range assets {
+		require.NoError(t, asset.Error)
+		num++
+	}
+	require.Equal(t, 0, num)
+
+	// Test 3: search for asset with results serialized properly
+	assets, _ = db.Assets(context.Background(), idb.AssetsQuery{Name: nameWithNull})
+	num = 0
+	var creator types.Address
+	for asset := range assets {
+		require.NoError(t, asset.Error)
+		require.Equal(t, nameWithNull, asset.Params.AssetName)
+		require.Equal(t, unitWithNull, asset.Params.UnitName)
+		require.Equal(t, urlWithNull, asset.Params.URL)
+		copy(creator[:], asset.Creator[:])
+		num++
+	}
+	require.Equal(t, 1, num)
+
```

Comment:
I had one more test in my branch that throws a wrench in things a little bit. The user isn't actually going to be able to search for `nameWithNull`, or invalid UTF8, so the main search parameter would be something like this:

```
	assets, _ = db.Assets(context.Background(), idb.AssetsQuery{Name: encoding.EncodeStringForQuery(nameWithNull)})
	num = 0
	for asset := range assets {
		require.NoError(t, asset.Error)
		require.Equal(t, encodedNameWithNull, asset.Params.AssetName)
		require.Equal(t, encodedUnitWithNull, asset.Params.UnitName)
		require.Equal(t, encodedUrlWithNull, asset.Params.URL)
		num++
	}
	require.Equal(t, 1, num)
```

I'm not sure we can support this and `idb.AssetsQuery{Name: nameWithNull}`